### ISSUE TITLE
ecs-cli ps --cluster <cluster-name> error: (FATA[0120] Error executing )

### ISSUE THREAD

**Issue by bugre** [aws/amazon-ecs-cli#984]
#### Summary

OS: Mac Osx 10.14.6
ecs-cli version 1.18.0 (3970a6c)
No 'go' installed.

command i'm trying to run `ecs-cli ps --cluster <aws_cluster_name>` get error:

```
FATA[0120] Error executing 'ps': RequestError: send request failed
caused by: Post https://ecs.eu-west-1.amazonaws.com/: dial tcp: i/o timeout
```

#### Description
ecs-cli used to work without problems using the aws-cli default profile/credentials. 

I use ecs-cli, mostly to list my tasks from a specific cluster and get it's IP's and status.

Credential/Profile haven't changed and seem to be correct, as command like `aws ecs list-tasks  --cluster <aws_cluster_name>` work fine.
```
werner-mbp:~ $ aws ecs list-tasks  --cluster <aws_cluster_name>
{
    "taskArns": [
        "arn:aws:ecs:eu-west-1:xx:task/xx1",
        "arn:aws:ecs:eu-west-1:xx:task/xx2",
        "arn:aws:ecs:eu-west-1:xx:task/xx3",
        "arn:aws:ecs:eu-west-1:xx:task/xx4"
    ]
}
```

nor can i remember any other significant changes, but could always be. I've reinstalled ecs-cli from source using the binary provided, also tried installing it with `brew` but always with the same result.

The errors refers to a network error apparently, but other services are working fine on my laptop.

```
werner-mbp:~ $ ecs-cli --version
ecs-cli version 1.18.0 (3970a6c)

werner-mbp:~ $ ecs-cli ps --cluster <aws_cluster_name>
FATA[0120] Error executing 'ps': RequestError: send request failed
caused by: Post https://ecs.eu-west-1.amazonaws.com/: dial tcp: i/o timeout
```

**testing the endpoint with curl, seems to work as expected, i get 403 as i'm not authenticated**
```
werner-mbp:~ $ curl -Iv 'https://ecs.eu-west-1.amazonaws.com/'
*   Trying 52.95.116.181:443...
* TCP_NODELAY set
* Connected to ecs.eu-west-1.amazonaws.com (52.95.116.181) port 443 (#0)
* ALPN, offering http/1.1
* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
* ALPN, server did not agree to a protocol
* Server certificate: ecs.eu-west-1.amazonaws.com
* Server certificate: Amazon
* Server certificate: Amazon Root CA 1
> HEAD / HTTP/1.1
> Host: ecs.eu-west-1.amazonaws.com
> User-Agent: curl/7.68.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 403 Forbidden
HTTP/1.1 403 Forbidden
< x-amzn-RequestId: 2bb29eeb-a883-4992-a46e-a832501d3d67
x-amzn-RequestId: 2bb29eeb-a883-4992-a46e-a832501d3d67
< Content-Length: 127
Content-Length: 127
< Date: Mon, 17 Feb 2020 13:13:26 GMT
Date: Mon, 17 Feb 2020 13:13:26 GMT
```

Any idea what i'm missing?
Thanks.

**Comment by efekarakus** [comment]
Hi @bugre,

I'm still investigating the error but I can't reproduce it :(   
At first glance this seems to come from the AWS SDK. There were previous issues cut with the same error and it seems to fix itself somehow 🤔 https://github.com/aws/amazon-ecs-cli/issues/76

Can you try to the following:
* Specify an explicit region with the `--region` flag?
* Try using the command over a different network?

In the mean time, I'm going to send a PR to upgrade the version of the SDK used in the CLI.

**Comment by efekarakus** [comment]
https://github.com/aws/amazon-ecs-cli/pull/985

**Comment by bugre** [comment]
Hi @efekarakus ,

Thanks for taking a look at the problem i'm facing.

Just did try it with the --region flag, with the same result :(
![image](https://user-images.githubusercontent.com/6900672/74688291-dbfddf80-51b5-11ea-84b3-84f0aa212c72.png)

Unfortunately at the moment i'm unable to test it with another network, but for now i would think that the network is find, as the 'aws cli' to the same region and the curl to the endpoint that ecs-cli complains work without problems.

I'll hope that the problem disappears agains without major problemas. 

Thanks
-wm

**Comment by bugre** [comment]
It's working fine again :| 
After further researching my problem, it seems that there was some DNS error involved, but i can't find why it only affected ecs-cli, and not anything different, like ping, curl, awscli even when connecting agains the same ecs-cli endpoint. 
I came to this , network sniffing the ecs-cli ps call and, it seemed to be making a call to resolve the endpoint name and never getting an answer, the rare, was that curl or ping did the same 'resolver' request and they got the answer immediately.

As it's working now, i stopped looking into it

**Comment by efekarakus** [comment]
Thanks for letting us know!

**Comment by patrickacioli** [comment]
> Hi @bugre,
> 
> I'm still investigating the error but I can't reproduce it :(
> At first glance this seems to come from the AWS SDK. There were previous issues cut with the same error and it seems to fix itself somehow  #76
> 
> Can you try to the following:
> 
> * Specify an explicit region with the `--region` flag?
> * Try using the command over a different network?
> 
> In the mean time, I'm going to send a PR to upgrade the version of the SDK used in the CLI.

The region flag (--region), works for me.
